recipes/cppcheck/all/conanfile.py

```diff
@@ -0,0 +1,66 @@
+"""Conan recipe package for cppcheck
+"""
+import os
+from conans import ConanFile, CMake, tools
+
+
+class CppcheckConan(ConanFile):
+    name = "cppcheck"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/danmar/cppcheck"
+    topics = ("Cpp Check", "static analyzer")
+    description = "Cppcheck is an analysis tool for C/C++ code."
+    license = "GPL-3.0-or-later"
+    generators = {"cmake"}
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"with_z3": [True, False], "have_rules": [True, False]}
+    default_options = {"with_z3": True, "have_rules": True}
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+        
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+    def requirements(self):
+        if self.options.with_z3:
+            self.requires("z3/4.8.8")
+        if self.options.have_rules:
+            self.requires("pcre/8.44")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "cppcheck-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
```

Comment:
Usually we ask to cache the CMake object, example: https://github.com/conan-io/conan-center-index/pull/3086/files#diff-ec045c054e4b94f8729715c3ec798f88b6dbda611dea1dd091597e2fa3d615bfR24

Reply:
Fixed